### PR TITLE
Add LTI Provider to the AskUp project

### PR DESCRIPTION
**Pull request by idegtiarov**
Add new django app `askup_lti` to the askup project. Askup_lti adds LTI provider with required infrastructure to the project and gives a possibility to add qset into LMS Platform via LTI component.

LTI launch url should be constructed as `<host>/lti/launch/qset/<qset_id>` for example:
`https://askup.com/lti/launch/qset/4`

### CODE REVIEW

File: askup_lti/models.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        """
+        return '<LtiProvider: {}>'.format(self.consumer_name)
+
+
+class LtiUser(models.Model):
+    """
+    Model to manage LTI users.
+    """
+
+    user_id = fields.CharField(max_length=255)
```
Comment (realARCHaim) [inline askup_lti/models.py:51]: Why not to use simple "id" here?
user_id is used a lot in other models of the project to link to the  Askup User.

File: askup_lti/provider.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        tool_provider = DjangoToolProvider.from_django_request(request=request)
+        validator = SignatureValidator()
+        ok = tool_provider.is_valid_request(validator)
+    except (oauth1.OAuth1Error, InvalidLTIRequestError, ValueError) as err:
+        ok = False
+        log.error('Error happened while LTI request: {}'.format(err.__str__()))
+    if not ok:
+        raise Http404('LTI request is not valid')
+
+    anononcement_page = render(
```
Comment (realARCHaim) [inline askup_lti/provider.py:37]: announcement?

File: askup_lti/models.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        :param organization: Organization object to check registration.
+        :return: boolean flag.
+        """
+        return self.askup_user.qset_set.filter(pk=organization.pk).exists()
+
+    def lti_to_askup_user(self):
+        """
+        Connect LTI user with the AskUp user account.
+        """
+        askup_user, _ = User.objects.get_or_create(username=self.user_id)
```
Comment (realARCHaim) [inline askup_lti/models.py:87]: Why don't you assign a role to the askup user here?

**Review by realARCHaim — CHANGES_REQUESTED** [review]

File: askup_lti/models.py (hunk 1 above)
Comment (idegtiarov) [reply askup_lti/models.py:51]: It is not a general id field it is identification which is received from the LTI request

File: askup_lti/provider.py (hunk 2 above)
Comment (idegtiarov) [reply askup_lti/provider.py:37]: Don't you like word `page`? ))

File: askup_lti/models.py (hunk 3 above)
Comment (idegtiarov) [reply askup_lti/models.py:87]: Nice catch! Thanks.

**Comment by elliottyates** [comment]
👏 @idegtiarov and @KhaimovMR

**Pull request merged**